blog/go-provider-sdk/index.mdx

```diff
@@ -0,0 +1,346 @@
+)
+
+func main() {
+	if err := run(); err != nil {
+		log.Fatal(err)
+	}
+}
+
+func run() error {
+	// Initialize the provider with callbacks to track linked components
+	providerHandler := Handler{}
+	p, err := provider.New(
+		provider.SourceLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewSourceLink(&providerHandler, link)
+		}),
+		provider.TargetLinkPut(func(link provider.InterfaceLinkDefinition) error {
+			return handleNewTargetLink(&providerHandler, link)
+		}),
+		provider.SourceLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelSourceLink(&providerHandler, link)
+		}),
+		provider.TargetLinkDel(func(link provider.InterfaceLinkDefinition) error {
+			return handleDelTargetLink(&providerHandler, link)
+		}),
+		provider.HealthCheck(func() string {
+			return handleHealthCheck(&providerHandler)
+		}),
+		provider.Shutdown(func() error {
+			return handleShutdown(&providerHandler)
+		}),
+	)
+	if err != nil {
+		return err
+	}
+
+	// Store the provider for use in the handlers
+	providerHandler.provider = p
+
+	// Setup two channels to await RPC and control interface operations
+	providerCh := make(chan error, 1)
+	signalCh := make(chan os.Signal, 1)
+
+	// Handle RPC operations
+	stopFunc, err := server.Serve(p.RPCClient, &providerHandler)
+	if err != nil {
+		p.Shutdown()
+		return err
+	}
+
+	// Handle control interface operations
+	go func() {
+		err := p.Start()
+		providerCh <- err
+	}()
+
+	// Shutdown on SIGINT
+	signal.Notify(signalCh, syscall.SIGINT)
+
+	// Run provider until either a shutdown is requested or a SIGINT is received
+	select {
+	case err = <-providerCh:
+		stopFunc()
+		return err
+	case <-signalCh:
+		p.Shutdown()
+		stopFunc()
+	}
+
+	return nil
+}
+
+func handleNewSourceLink(handler *Handler, link provider.InterfaceLinkDefinition) error {
+	handler.provider.Logger.Info("Handling new source link", "link", link)
+	handler.linkedTo[link.SourceID] = link.SourceConfig
+	return nil
+}
+
+func handleNewTargetLink(handler *Handler, link provider.InterfaceLinkDefinition) error {
+	handler.provider.Logger.Info("Handling new target link", "link", link)
+	handler.linkedFrom[link.Target] = link.TargetConfig
+	return nil
+}
+
+func handleDelSourceLink(handler *Handler, link provider.InterfaceLinkDefinition) error {
+	handler.provider.Logger.Info("Handling del source link", "link", link)
+	delete(handler.linkedTo, link.SourceID)
+	return nil
+}
+
+func handleDelTargetLink(handler *Handler, link provider.InterfaceLinkDefinition) error {
+	handler.provider.Logger.Info("Handling del target link", "link", link)
+	delete(handler.linkedFrom, link.Target)
+	return nil
+}
+
+func handleHealthCheck(handler *Handler) string {
+	handler.provider.Logger.Info("Handling health check")
+	return "provider healthy"
+}
+
+func handleShutdown(handler *Handler) error {
+	handler.provider.Logger.Info("Handling shutdown")
+	clear(handler.linkedFrom)
+	clear(handler.linkedTo)
+	return nil
```

Comment:
These just need the println fix from https://github.com/wasmCloud/wasmCloud/pull/2386